agent/bench-scripts/pbench-linpack

```diff
@@ -138,16 +166,37 @@ fi
 
 trap "interrupt" INT QUIT TERM
 
-## Run the benchmark and start/stop perf analysis tools
-pbench-start-tools --group=$tool_group --dir=$benchmark_results_dir
-
-OMP_NUM_THREADS=$threads ${linpack_dir}/${linpack_cmd} < ${linpack_dir}/${linpack_dat} | tee $result_file
-
-pbench-stop-tools --group=$tool_group --dir=$benchmark_results_dir
-pbench-send-tools --group=$tool_group --dir=$benchmark_results_dir
-pbench-postprocess-tools --group=$tool_group --dir=$benchmark_results_dir
-
-ln -s ${sample} $benchmark_run_dir/$iteration/reference-result
+let count=1
+for thread in ${threads//,/ }; do
+	iteration="$count-$thread-threads"
+	echo $iteration >> $benchmark_iterations
+	echo "Starting iteration ${iteration}"
+	# Generate the linpack.input dirs (in /var/lib/pbench-agent/tmp)
+	iteration_dir="$benchmark_run_dir/$iteration"
+	mkdir -p "$iteration_dir"
+	store_and_run "$iteration_dir" "'$pbench_bin/bench-scripts/postprocess/linpack-prepare-input-file' --output-dir '$iteration_dir' --threads '$thread' --run-samples 1 --linpack-binary '$linpack_dir/$linpack_cmd'" "$benchmark-preprocess.cmd"
+	for sample in $(seq 1 $nr_samples); do
+		echo "test sample $sample of $nr_samples"
+		sample_dir="$iteration_dir/sample$sample"
+		# Create the output directory expected by the postprocess script
+		mkdir -p "$sample_dir/clients/localhost/"
+
+		## Run the benchmark and start/stop perf analysis tools
+		pbench-start-tools --group=$tool_group --dir="$sample_dir"
+
+		( cd "$iteration_dir" && ./linpack.sh )
+
+		pbench-stop-tools --group=$tool_group --dir="$sample_dir"
+		pbench-send-tools --group=$tool_group --dir="$sample_dir"
+		# Move the results to the output directory expected by the postprocess script
+		mv "$iteration_dir/"linpack.{out,meta} "$benchmark_results_dir/clients/localhost"
```

Comment:
Where is `$benchmark_results_dir` defined?  (And, what is it defined to?)

Previously, it was defined to `"$benchmark_run_dir/$iteration/${sample}"`.  I think this would now be equivalent to `"$sample_dir"` (which is supported by the various references to it in lines 185-194).

So, it looks like line 192 should be
```
		mv "$iteration_dir/"linpack.{out,meta} "$sample_dir/clients/localhost"
```
which would match line 182.

Am I missing something?

Reply:
Oups, my bad. I forgot to change this in the `Unify output dir naming in pbench-linpack` commit. Addressed now.

Reply:
(I only tested it before renaming the variables, it was just a mass rename, what could happen, right?)